common/iterator.go

```diff
@@ -18,16 +18,14 @@ import (
 	"context"
 
 	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-connector-sdk"
 )
 
-// Iterator is an object that can iterate over a queue of records.
+// Iterator implements a subset of the sdk.Source interface.
 type Iterator interface {
-	// Read takes and returns the next record from the queue. Read is allowed to
-	// block until either a record is available or the context gets canceled.
-	Next(context.Context) (opencdc.Record, error)
-	// Ack signals that a record at a specific position was successfully
-	// processed.
+	Read(context.Context) (opencdc.Record, error)
 	Ack(context.Context, opencdc.Position) error
-	// Teardown attempts to gracefully teardown the iterator.
 	Teardown(context.Context) error
 }
+
+var _ Iterator = (sdk.Source)(nil)
```

Comment:
It's a bit confusing. The line actually says that the `sdk.Source` is an `Iterator`, not the other way around (I had to think a bit to write this, wasn't obvious, which was the intent of this line).

I wrote this so that there's a clear naming pattern when calling iterators. The root iterator, Source, calls `Read()` on the child iterators. In integration tests this makes it easier to make utilities, as you can see in `testutils`. I can pass a source, a snapshot iterator or a cdc iterator in the utilities. 

This ensures that the naming is right. Before this line I had a custom Iterator with a `Next()` method, so in all the test utilities I had to do an annoying wrapper for the source struct.

I personally do like it, it's the closest `implements` java thing that go has.